Graphs/KahnsAlgorithm.js

```diff
@@ -0,0 +1,45 @@
+import Queue from '../Data-Structures/Queue/Queue'
+
+/**
+ * @author {RaviSadam}
+ * @name KahnsAlgorithm
+ * @description -
+ * Kahn's Algorithm implementation in JavaScript
+ * @summary
+ * Kahn's Algorithm is used for finding the topological sorting order of directed acyclic graph
+ *
+ * @param graph - Graph (adjacency list)
+ * @param n - Size of graph
+ * @returns {Array} - null if cycle is detected or else result array
+ *
+ */
+
+export function KahnsAlgorithm(graph, n) {
+  if (n === null || n === undefined) throw Error('Invalid n was given')
+  const inDegree = Array(n).fill(0)
+  const result = []
+  for (const neighbours of graph) {
+    for (const neighbour of neighbours) {
+      inDegree[neighbour] += 1
+    }
+  }
+  const queue = new Queue()
```

Comment:
Why use a queue when you can get away with just using JS data structures, specifically, a stack?

Also, why call this "queue" rather than giving it a more descriptive name like "roots"?